executor/analyze_test.go

```diff
@@ -234,14 +264,15 @@ func (s *testFastAnalyze) TestAnalyzeFastSample(c *C) {
 	opts[ast.AnalyzeOptNumSamples] = 20
 	mockExec := &executor.AnalyzeTestFastExec{
 		Ctx:             tk.Se.(sessionctx.Context),
-		PKInfo:          pkCol,
+		HandleCols:      handleCols,
 		ColsInfo:        colsInfo,
 		IdxsInfo:        indicesInfo,
 		Concurrency:     1,
 		PhysicalTableID: tbl.(table.PhysicalTable).GetPhysicalID(),
 		TblInfo:         tblInfo,
 		Opts:            opts,
 	}
+	tk.MustExec("begin")
```

Comment:
Why need the transaction?

Reply:
This is used in `calculateEstimateSampleStep`:
```go
txn, err := e.ctx.Txn(true)
if err != nil {
	return 0, errors.Trace(err)
}
txn.SetOption(kv.Priority, kv.PriorityLow)
txn.SetOption(kv.IsolationLevel, kv.RC)
txn.SetOption(kv.NotFillCache, true)
sql := fmt.Sprintf("select count(*) from %s.%s;", dbInfo.Name.L, e.tblInfo.Name.L)
recordSets, err := e.ctx.(sqlexec.SQLExecutor).ExecuteInternal(context.TODO(), sql)
```

Reply:
The test here invokes the method `buildStats()` in the fast-analyze executor directly. If we don't `begin`, it reports 'invalid transaction'.

Reply:
Updated. Now it will run 'begin/rollback' internally if the transaction is invalid.